src/modules/internet/libcurl.c

```diff
@@ -236,35 +236,25 @@ static void curlCommon(CURL *hnd, int redirect, int verify)
 	curl_easy_setopt(hnd, CURLOPT_SSL_VERIFYHOST, 0L);
 	curl_easy_setopt(hnd, CURLOPT_SSL_VERIFYPEER, 0L);
     }
-#if 0
     // for consistency, but all utils:::makeUserAgent does is look up an option.
     SEXP sMakeUserAgent = install("makeUserAgent");
     SEXP agentFun = PROTECT(lang2(sMakeUserAgent, ScalarLogical(0)));
     SEXP utilsNS = PROTECT(R_FindNamespace(mkString("utils")));
     SEXP sua = eval(agentFun, utilsNS);
     UNPROTECT(1); /* utilsNS */
     PROTECT(sua);
-    if(TYPEOF(sua) != NILSXP)
-	curl_easy_setopt(hnd, CURLOPT_USERAGENT, CHAR(STRING_ELT(sua, 0)));
+
+    /* Prefix with libcurl version, always */
+    curl_version_info_data *d = curl_version_info(CURLVERSION_NOW);
+    int sualen = isNull(sua) ? 0 : strlen(CHAR(STRING_ELT(sua, 0)));
+    char *useragent = R_alloc(sualen + 30, 1);
+    snprintf(useragent, sualen + 30,
+             "libcurl/%s%s%s", d->version, sualen ? ", " : "",
```

Comment:
I'm not sure if this is really standardized, but I usually see multiple user agents separated with a space. For example, in https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent and https://tools.ietf.org/html/rfc7231#section-5.5.3. Should we follow that convention and use a space instead of a comma?

Reply:
Sure.